lib/elixir/lib/inspect.ex

```diff
@@ -380,21 +380,38 @@ defimpl Inspect, for: Map do
 end
 
 defimpl Inspect, for: Integer do
+  def inspect(term, %Inspect.Opts{base: :decimal}) do
+    Integer.to_string(term)
+    |> pretty_decimal
+  end
+
   def inspect(term, %Inspect.Opts{base: base}) do
     Integer.to_string(term, base_to_value(base))
     |> prepend_prefix(base)
   end
 
+  def pretty_decimal(string) do
+    split_index = 
+      case rem(byte_size(string), 3) do
+        0 -> 3
+        x -> x
+      end
+    pretty_decimal(string, split_index)
+  end
+  def pretty_decimal(string, _) when byte_size(string) <= 3, do: string
+  def pretty_decimal(string, split_index) do
+     {left, right} = String.split_at(string, split_index)
+     left <> "_" <> pretty_decimal(right, 3)
+  end
```

Comment:
What if we do this before converting the integer to a string? E.g.

```elixir
def pretty_decimal(n) when n >= 1_000 do
  left = div(n, 1_000)
  right = rem(n, 1_000)
  pretty_decimal(left) <> "_" <> Integer.to_string(right)
end

def pretty_decimal(n) do
  Integer.to_string(n)
end
```

The implementation looks slightly easier to read maybe; I haven't tested it so it may not work at all :)